cmd/kube-rbac-proxy/app/setup_authorizer_test.go

```diff
@@ -0,0 +1,430 @@
+}
+
+func TestSetupAuthorizer_AllowPathsWithRewriteAndStaticAuth(t *testing.T) {
+	mockDelegated := mockAuthorizer(
+		func(ctx context.Context, a authorizer.Attributes) (authorizer.Decision, string, error) {
+			if a.GetUser().GetName() == "system:serviceaccount:default:client-with-rbac" {
+				return authorizer.DecisionAllow, "delegated allow", nil
+			}
+			return authorizer.DecisionNoOpinion, "i don't care", nil
+		},
+	)
+
+	mockErr := mockAuthorizer(
+		func(ctx context.Context, a authorizer.Attributes) (authorizer.Decision, string, error) {
+			return authorizer.DecisionDeny, "delegated deny", fmt.Errorf("delegated error")
+		},
+	)
+
+	krbInfo := &server.KubeRBACProxyInfo{
+		AllowPaths: []string{"/metrics"},
+		Authorization: &authorization.AuthzConfig{
+			RewriteAttributesConfig: &rewrite.RewriteAttributesConfig{
+				Rewrites: &rewrite.SubjectAccessReviewRewrites{
+					ByHTTPHeader: &rewrite.HTTPHeaderRewriteConfig{
+						Name: "x-namespace",
+					},
+				},
+				ResourceAttributes: &rewrite.ResourceAttributes{
+					Resource:  "namespaces",
+					Namespace: "{{ .Value }}",
+				},
+			},
+			Static: []static.StaticAuthorizationConfig{
+				{
+					User: static.UserConfig{
+						Name: "system:serviceaccount:default:client-with-static",
+					},
+					ResourceRequest: true,
+					Resource:        "namespaces",
+					Namespace:       "kube-system",
+					Verb:            "get",
+				},
+			},
+		},
+	}
+
+	errStatic := &server.KubeRBACProxyInfo{
+		AllowPaths: []string{"/metrics"},
+		Authorization: &authorization.AuthzConfig{
+			RewriteAttributesConfig: &rewrite.RewriteAttributesConfig{
+				Rewrites: &rewrite.SubjectAccessReviewRewrites{
+					ByHTTPHeader: &rewrite.HTTPHeaderRewriteConfig{
+						Name: "x-namespace",
+					},
+				},
+				ResourceAttributes: &rewrite.ResourceAttributes{
+					Resource:  "namespaces",
+					Namespace: "{{ .Value }}",
+				},
+			},
+			Static: []static.StaticAuthorizationConfig{
+				{
+					User: static.UserConfig{
+						Name: "system:serviceaccount:default:client-with-static",
+					},
+					ResourceRequest: false,
+					Resource:        "namespaces",
+					Namespace:       "kube-system",
+					Verb:            "get",
+				},
+			},
+		},
+	}
+
+	errPath := &server.KubeRBACProxyInfo{
+		AllowPaths: []string{"/foo/*/metrics"},
+		Authorization: &authorization.AuthzConfig{
+			RewriteAttributesConfig: &rewrite.RewriteAttributesConfig{
+				Rewrites: &rewrite.SubjectAccessReviewRewrites{
+					ByHTTPHeader: &rewrite.HTTPHeaderRewriteConfig{
+						Name: "x-namespace",
+					},
+				},
+				ResourceAttributes: &rewrite.ResourceAttributes{
+					Resource:  "namespaces",
+					Namespace: "{{ .Value }}",
+				},
+			},
+			Static: []static.StaticAuthorizationConfig{
+				{
+					User: static.UserConfig{
+						Name: "system:serviceaccount:default:client-with-static",
+					},
+					Resource:  "namespaces",
+					Namespace: "kube-system",
+					Verb:      "get",
+				},
+			},
+		},
+	}
```

Comment:
don't make people try to diff these two complex objects when they get here, have a base where you supply/modify whatever necessary